components/ILIAS/UI/tests/Component/Layout/Page/MailPageTest.php

```diff
@@ -0,0 +1,202 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+namespace Component\Layout\Page;
+
+require_once('vendor/composer/vendor/autoload.php');
+require_once(__DIR__ . '/../../../Base.php');
+
+use ILIAS\UI\Implementation\Component\Legacy\Content;
+use ILIAS_UI_TestBase;
+use ILIAS\UI\Implementation\Component\Layout\Page;
+
+class MailPageTest extends ILIAS_UI_TestBase
+{
+    protected Page\Mail $mailpage;
+    protected Page\Factory $factory;
+    protected string $logo;
+    protected string $installation_title;
+    protected string $stylesheet_path;
+    protected Content $html_content;
+    protected string $footer_url;
+
+    public function setUp(): void
+    {
+        $this->html_content = $this->createMock(Content::class);
+        $this->html_content->method('getContent')->willReturn('ILIAS HTML Content');
+
+        $this->logo = 'ILIAS Logo';
+        $this->installation_title = 'ILIAS Installation Title';
+        $this->stylesheet_path = __FILE__;
+        $this->footer_url = 'ILIAS/footer/url';
+
+        $this->factory = new Page\Factory();
+        $this->mailpage = $this->factory->mail(
+            $this->html_content,
+            $this->logo,
+            $this->installation_title,
+            $this->footer_url,
+            $this->stylesheet_path
+        );
+    }
+
+    public function testConstruction(): void
+    {
+        static::assertInstanceOf(
+            Page\Mail::class,
+            $this->mailpage
+        );
+    }
+
+    public static function provideGetData(): array
+    {
+        return [
+            ['html_content', 'getContent', 'assertContains'],
+            ['logo', 'getLogoSrc', 'assertEquals'],
+            ['installation_title', 'getInstallationTitle', 'assertEquals'],
+            ['stylesheet_path', 'getStyleSheetPath', 'assertEquals'],
+            ['footer_url', 'getFooterURL', 'assertEquals'],
+        ];
+    }
+
+    /**
+     * @dataProvider provideGetData
+     */
+    public function testGet(mixed $expected, string $method_to_call, string $assert_method): void
+    {
+        static::$assert_method(
+            $this->$expected,
+            call_user_func([$this->mailpage, $method_to_call])
```

Comment:
`call_user_func` and `call_user_func_array` can be called directly without the need for those functions.

```suggestion
            $this->mailpage->$method_to_call()
```